### PR TITLE
Remove CHIP_CONFIG_SENDING_BATCH_COMMANDS_ENABLED

### PR DESCRIPTION
**Pull request by tehampson**
Now that we have `CommandSender::ExtendableCallback`, we can rely on that to make sure anything using `CommandSender` is capable of sending batch commands

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

File: src/app/CommandSender.h
Comment (bzbarsky-apple) [inline src/app/CommandSender.h:307]:
```suggestion
     *                      CommandSender has not been provided callbacks expected for supporting batch commands.
```

File: src/app/CommandSender.cpp
Comment (bzbarsky-apple) [inline src/app/CommandSender.cpp:415]: So this was not a "reduce codesize" kind of ifdef?

Comment (tehampson) [reply src/app/CommandSender.cpp:415]: Right now it doesn't seem to be doing much based on the size comparison report. With the exception for on target on `bl702l`, which seems like read and write section increased by 500 bytes, while having a decrease in code size (text).

We could be more aggressive in using this ifdef to reduce code usage. Although historically this was initially added in case there was anything important to block on this, but with the latest addition of `CommandSender::ExtendableCallback`, which is something we want people to move to regardless of batch commands, the needs for `CHIP_CONFIG_SENDING_BATCH_COMMANDS_ENABLED` seems to be diminishing.

With that all that said, Tennessee did mention that `CommandSender` should be doing more than it is today for tracking which commands have come back. So I will hold off on merging this PR until that lands and we can see if there is a code size reduction worth keeping around

Comment (tehampson) [reply src/app/CommandSender.cpp:415]: Additionally one reason that I wanted to remove this `CHIP_CONFIG_SENDING_BATCH_COMMANDS_ENABLED` was to remove some test gating on cleaning up some unit tests to use CommandSender to build batch commands

https://github.com/project-chip/connectedhomeip/pull/31081

**Comment by tehampson** [comment]
With https://github.com/project-chip/connectedhomeip/pull/32105, this CONFIG is now required

**Pull request closed without merging**
